File: frontend/src/api/tenant.ts
import client from './client'

export interface Tenant {
  id: string
  code: string
  name: string
  description?: string
  status: boolean
  created_at: string
  updated_at: string
  created_by?: string
}

export interface TenantCreate {
  code: string
  name: string
  description?: string
}

export interface TenantUpdate {
  code?: string
  name?: string
  description?: string
  status?: boolean
}

export const tenantApi = {
  // 获取租户列表
  getList: async (params?: { skip?: number; limit?: number; status?: boolean }) => {
    const response = await client.get<Tenant[]>('/tenants', { params })
    return response.data
  },

  // 获取单个租户
  getById: async (id: string) => {
    const response = await client.get<Tenant>(`/tenants/${id}`)
    return response.data
  },

  // 创建租户
  create: async (data: TenantCreate) => {
    const response = await client.post<Tenant>('/tenants', data)
    return response.data
  },

  // 更新租户
  update: async (id: string, data: TenantUpdate) => {
    const response = await client.put<Tenant>(`/tenants/${id}`, data)
    return response.data
  },

  // 删除租户
  delete: async (id: string) => {
    await client.delete(`/tenants/${id}`)
  },

  // 启用/停用租户
  updateStatus: async (id: string, status: boolean) => {
    const response = await client.patch<Tenant>(`/tenants/${id}/status`, null, {
      params: { status },
    })
    return response.data
  },
}

File: app/schemas/tenant.py
"""
租户Schema
"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class TenantBase(BaseModel):
    """租户基础Schema"""
    code: str = Field(..., min_length=1, max_length=50, description="租户编码")
    name: str = Field(..., min_length=1, max_length=100, description="租户名称")
    description: Optional[str] = Field(None, max_length=500, description="描述")


class TenantCreate(TenantBase):
    """创建租户Schema"""


class TenantUpdate(BaseModel):
    """更新租户Schema"""
    code: Optional[str] = Field(None, min_length=1, max_length=50, description="租户编码")
    name: Optional[str] = Field(None, min_length=1, max_length=100, description="租户名称")
    description: Optional[str] = Field(None, max_length=500, description="描述")
    status: Optional[bool] = Field(None, description="状态：True-启用，False-停用")


class TenantResponse(TenantBase):
    """租户响应Schema"""
    id: str
    status: bool
    created_at: datetime
    updated_at: datetime
    created_by: Optional[str] = None
    
    class Config:
        from_attributes = True

